app/git/utils.py

```diff
@@ -277,6 +277,41 @@ def get_github_primary_email(oauth_token):
     return ''
 
 
+def get_github_event_emails(oauth_token, username):
+    """Get all email addresses associated with the github profile.
+
+    Args:
+        oauth_token (str): The Github OAuth2 token to use for authentication.
+
+    Returns:
+        list of str: All of the user's associated email from github.
+
+    """
+    emails = []
+    headers = JSON_HEADER
+    if oauth_token:
+        headers = dict({'Authorization': f'token {oauth_token}'}, **JSON_HEADER)
+    response = requests.get(f'https://api.github.com/users/{username}/events/public', headers=headers)
+
+    userinfo = get_user(username)
+    user_name = userinfo.get('name', '')
+    print(user_name)
+
+    if response.status_code == 200:
+        events = response.json()
+        for event in events:
+            payload = event.get('payload', {})
+            for commit in payload.get('commits', []):
+                author = commit.get('author', {})
+                email = author.get('email', {})
+                name = author.get('name', {})
+                if name.lower() == username.lower() or name.lower() == user_name.lower():
+                    if email and 'noreply.github.com' not in email:
```

Comment:
I think this can be combined into a single `if`?

Reply:
👍